packages/schematics/angular/utility/workspace-models.ts

```diff
@@ -0,0 +1,150 @@
+    scripts?: string[];
+    sourceMap?: boolean;
+}
+
+export interface BrowserBuilderOptions extends BrowserBuilderBaseOptions {
+    serviceWorker?: boolean;
+    optimization?: boolean;
+    outputHashing?: 'all';
+    extractCss?: boolean;
+    namedChunks?: boolean;
+    aot?: boolean;
+    extractLicenses?: boolean;
+    vendorChunk?: boolean;
+    buildOptimizer?: boolean;
+}
+
+export interface ServeBuilderOptions {
+    browserTarget: string;
+}
+export interface LibraryBuilderOptions {
+    tsConfig: string;
+    project: string;
+}
+
+export interface ServerBuilderOptions {
+    outputPath: string;
+    tsConfig: string;
+    main: string;
+    fileReplacements?: FileReplacements[];
+}
+
+export interface AppShellBuilderOptions {
+    browserTarget: string;
+    serverTarget: string;
+    route: string;
+}
+
+export interface TestBuilderOptions extends Partial<BrowserBuilderBaseOptions> {
+    karmaConfig: string;
+}
+
+export interface LintBuilderOptions {
+    tsConfig: string[] | string;
+    exclude?: string[];
+}
+
+export interface ExtractI18nOptions {
+    browserTarget: string;
+}
+
+export interface E2EOptions {
+    protractorConfig: string;
+    devServerTarget: string;
+}
+
+export interface BuilderTarget<TBuilder extends Builders, TOptions> {
+    builder: TBuilder;
+    options: TOptions;
+    configurations?: {
+        production: Partial<TOptions>;
+        [key: string]: Partial<TOptions>;
+    };
+}
+
+export type LibraryBuilderTarget = BuilderTarget<Builders.NgPackagr, LibraryBuilderOptions>;
+export type BrowserBuilderTarget = BuilderTarget<Builders.Browser, BrowserBuilderOptions>;
+export type ServerBuilderTarget = BuilderTarget<Builders.Server, ServerBuilderOptions>;
+export type AppShellBuilderTarget = BuilderTarget<Builders.AppShell, AppShellBuilderOptions>;
+export type LintBuilderTarget = BuilderTarget<Builders.TsLint, LintBuilderOptions>;
+export type TestBuilderTarget = BuilderTarget<Builders.Karma, TestBuilderOptions>;
+export type ServeBuilderTarget = BuilderTarget<Builders.DevServer, ServeBuilderOptions>;
+export type ExtractI18nBuilderTarget = BuilderTarget<Builders.ExtractI18n, ExtractI18nOptions>;
+export type E2EBuilderTarget = BuilderTarget<Builders.Protractor, E2EOptions>;
+
+export interface WorkspaceSchema extends experimental.workspace.WorkspaceSchema {
+    projects: {
+        [key: string]: WorkspaceProject<ProjectType.Application | ProjectType.Library>;
+    };
+}
+
+// tslint:disable-next-line:no-any
+export interface WorkspaceProject<TProjectType extends ProjectType = any>
+    extends experimental.workspace.WorkspaceProject {
+    /**
+    * Project type.
+    */
+    projectType: ProjectType;
+
+    /**
+     * Tool options.
+     */
+    architect?: WorkspaceTarget<TProjectType>;
+    /**
+     * Tool options.
+     */
+    targets?: WorkspaceTarget<TProjectType>;
+}
+
+export interface WorkspaceTarget<TProjectType extends ProjectType = ProjectType.Application> {
```

Comment:
This should be `WorkspaceTargets` (plural) where any of the keys are a target proper.